p/scripts/main.js

```diff
@@ -2097,15 +2092,18 @@ function init_normal() {
 	init_actualize();
 	faviconNbUnread();
 
-	window.onbeforeunload = function (e) {
-		const sidebar = document.getElementById('sidebar');
-		if (sidebar) {	// Save sidebar scroll position
-			sessionStorage.setItem('FreshRSS_sidebar_scrollTop', sidebar.scrollTop);
-		}
-		if (mark_read_queue && mark_read_queue.length > 0) {
-			return false;
+	document.addEventListener("visibilitychange", () => {
```

Comment:
I am quite in doubt that this will work to guarantee that the actions are completed.
Could we have a bit more information there, such as how it compares with `onbeforeunload`, and when it is fired / not fired?

Reply:
Just to be sure @Inverle , you are trying to address the case of https://github.com/FreshRSS/FreshRSS/issues/7410#issuecomment-2705411269 namely that some entries are in the process of getting marked as read (due to auto-marking-as-read during scroll) while the user navigates away?